str0m/src/media/mline.rs

```diff
@@ -991,6 +1001,55 @@ impl MLine {
         }
     }
 
+    /// Return the queue state for outbound RTP traffic for this mline.
+    ///
+    /// For the purposes of pacing each mline is treated as its own packet queue. Internally this
+    /// is spread across many [`PacketizingBuffer`] which is where the actual packets are queued.
+    pub fn send_queue_state(&self, now: Instant) -> QueueState {
+        let (size, queue_time, packet_count, audio_kind_sate, video_kind_state) =
+            self.buffers_tx.values().fold(
+                (
+                    DataSize::ZERO,
+                    Duration::ZERO,
+                    0,
+                    MediaState::default(),
+                    MediaState::default(),
+                ),
+                |mut acc, b| {
+                    (
+                        acc.0 + b.queue_size(),
+                        acc.1 + b.queue_time(now),
+                        acc.2 + b.queued_packet_count(),
+                        if b.is_audio() {
+                            acc.3.update(b.last_emit_time(), b.leading_queue_time());
+
+                            acc.3
+                        } else {
+                            acc.3
+                        },
+                        if !b.is_audio() {
+                            acc.4.update(b.last_emit_time(), b.leading_queue_time());
+
+                            acc.4
+                        } else {
+                            acc.4
+                        },
+                    )
+                },
+            );
+
+        QueueState {
+            queue_id: QueueId::new(self.index() as u64),
+            size,
+            packet_count: packet_count as u32,
+            queue_time,
+            media_states: MediaStates {
+                audio: audio_kind_sate,
+                video: video_kind_state,
```

Comment:
Why are these values split into audio/video? I understand we need that separation higher up, but an m-line can only be one of them. Wouldn't it be better to just have `media_state` and a flag `is_audio: bool`.

Reply:
It's a bit more future proof i.e. if we were to move to a single queue instead of one per mline the pacer code already handles it. This fact is utilized in the pacer tests where there's a single packet queue that may contain both audio and video

Reply:
Hmmm… "future proof" and "test reasons" are not very high on my agenda. It kinda confuses me because I expect these two be used at the same time, and they won't be.

Reply:
Okay I guess I can complicate the tests instead